biojava-structure/src/main/java/org/biojava/nbio/structure/secstruc/BasePairParameters.java

```diff
@@ -0,0 +1,403 @@
+                    "ATOM     12  C4    G A   1      -1.265   3.177   0.000\n" +
+                    "END",
+            "SEQRES   1 A    1  T\n" +
+                    "ATOM      2  N1    T A   1      -1.284   4.500   0.000\n" +
+                    "ATOM      3  C2    T A   1      -1.462   3.135   0.000\n" +
+                    "ATOM      5  N3    T A   1      -0.298   2.407   0.000\n" +
+                    "ATOM      6  C4    T A   1       0.994   2.897   0.000\n" +
+                    "ATOM      8  C5    T A   1       1.106   4.338   0.000\n" +
+                    "ATOM     10  C6    T A   1      -0.024   5.057   0.000\n" +
+                    "END",
+            "SEQRES   1 A    1  C\n" +
+                    "ATOM      2  N1    C A   1      -1.285   4.542   0.000\n" +
+                    "ATOM      3  C2    C A   1      -1.472   3.158   0.000\n" +
+                    "ATOM      5  N3    C A   1      -0.391   2.344   0.000\n" +
+                    "ATOM      6  C4    C A   1       0.837   2.868   0.000\n" +
+                    "ATOM      8  C5    C A   1       1.056   4.275   0.000\n" +
+                    "ATOM      9  C6    C A   1      -0.023   5.068   0.000\n" +
+                    "END"
+    };
+
+    private static String[] baseListDNA = {"A", "G", "T", "C"};
+    private static String[] baseListRNA = {"A", "G", "U", "C"};
+    private static Map<String, Integer> map;
+    private static Map<Integer, List<String>> ringMap;
+    static {
+        map = new HashMap<>();
+        map.put("DA", 0); map.put("ADE", 0); map.put("A", 0);
+        map.put("DG", 1); map.put("GUA", 1); map.put("G", 1);
+        map.put("DT", 2); map.put("THY", 2); map.put("T", 2); //RNA lines: map.put("U", 2); map.put("URA", 2);
+        map.put("DC", 3); map.put("CYT", 3); map.put("C", 3);
+        // chemically modified bases, leaving out right now.
+        //map.put("DZM", 0);
+        //map.put("UCL", 2);
+        //map.put("2DT", 2);
+        //map.put("1CC", 3); map.put("5CM", 3);
+        ringMap = new HashMap<>();
+        ringMap.put(0, Arrays.asList("C8", "C2", "N3", "C4", "C5", "C6", "N7", "N1", "N9"));
+        ringMap.put(1, Arrays.asList("C8", "C2", "N3", "C4", "C5", "C6", "N7", "N1", "N9"));
+        ringMap.put(2, Arrays.asList("C6", "C2", "N3", "C4", "C5", "N1"));
+        ringMap.put(3, Arrays.asList("C6", "C2", "N3", "C4", "C5", "N1"));
+   }
+
+    private Structure structure;
+    private String pairSequence = "", pdbId = "";
+
+    private double[] pairParameters;
+    private double[][] pairingParameters;
+    private double[][] stepParameters;
+
+
+    // Either pass the name of a PDB file (ending in .pdb) or pass the pdbId
+    public BasePairParameters(String name) {
+        PDBFileReader pdbFileReader = new PDBFileReader();
+        pdbFileReader.setPath(".");
+        try {
+            if (name.contains(".pdb")) structure = pdbFileReader.getStructure(name);
+            else structure = StructureIO.getStructure(name);
```

Comment:
Agreed.  the new version is using Structure directly, but this implementation had supported both using StructureIO and loading a local file, that second part is why the PDBFileReader was there.